core/src/slot_toggle.rs

```diff
@@ -66,7 +68,7 @@ impl SlotToggle {
     /// * `bool` - Whether the feature was successfully activated
     pub fn activate(&mut self, slot: u64, epoch_length: u64) -> bool {
         match self.state(slot, epoch_length) {
-            SlotToggleState::Inactive => {
+            Ok(SlotToggleState::Inactive) => {
```

Comment:
In these cases, we'll want to also check for the error and fail the program. So this function would also return a Result<bool, JsmCoreError>